model_api/cpp/models/src/segmentation_model.cpp

```diff
@@ -32,8 +32,80 @@
 #include "models/input_data.h"
 #include "models/results.h"
 
+namespace {
+cv::Mat create_hard_prediction_from_soft_prediction(const cv::Mat& soft_prediction, float soft_threshold, int blur_strength) {
```

Comment:
Why soft_threshold argument is not used here? Perhaps, thresholding makes no sense in most cases, since we have argmax, but in python you keep it.

Reply:
Yes, that's strange. It makes sense to return the most probable label, not the one with the smallest index if every labels has low probability for a given pixel. Can I completely drop soft_prediction? But this will introduce difference between OTX and ModelAPI